File: 백준/Silver/24479. 알고리즘 수업 － 깊이 우선 탐색 1/알고리즘 수업 － 깊이 우선 탐색 1.py
import sys
sys.setrecursionlimit(10**9)
input = sys.stdin.readline

n, m, r = map(int, input().split())

stack = [[] for _ in range(n+1)]
visited = [0] * (n + 1)
count = 1

for _ in range(m):
    u, v = map(int, input().split())
    stack[u].append(v)
    stack[v].append(u)

def DFS(v):
    global count
    visited[v] = count
    stack[v].sort()
    for i in stack[v]:
        if visited[i] == 0:
            count += 1
            DFS(i)


DFS(r)

for i in range(1, n + 1):
    print(visited[i])

File: 백준/Silver/24479. 알고리즘 수업 － 깊이 우선 탐색 1/알고리즘 수업 － 깊이 우선 탐색 1.kt
lateinit var graph: Array<MutableList<Int>>
lateinit var visited: IntArray
var cnt = 1

fun main() {
    val str = readln().split(" ").map { it.toInt() }
    val point = str[0]
    val node = str[1]
    var startPoint = str[2]
    graph = Array(point+1) {mutableListOf<Int>()}
    visited = IntArray(point + 1) {0}
    repeat(node) {
        val str = readln().split(" ").map { it.toInt() }
        val point = str[0]
        val node = str[1]
        graph[point].add(node)
        graph[node].add(point)
    }

    graph.map { it.sort() }

    dfs(startPoint)

    for(i in 1 until visited.size) {
        println(visited[i])
    }

}

fun dfs(start: Int) {
    if (visited[start] != 0) return
    visited[start] = cnt++
    graph[start].forEach {
        dfs(it)
    }
}